ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt

```diff
@@ -1,53 +1,171 @@
+ *  * "^" - alias symbol, can be used as follows:
+ *      1. In front of an import path, meaning matching all alias imports from this path, e.g. "^android.*"
+ *      2. In conjunction with "*" operator, meaning matching any alias import - "^*" (<all other alias imports> in IDEA)
+ *  * import paths - these can be full paths, e.g. "java.util.List" as well as wildcard paths, e.g. "kotlin.*"
  *
- * https://developer.android.com/kotlin/style-guide#import_statements
+ * In case the custom property is not provided, the rule defaults to "idea" style.
  */
 class ImportOrderingRule : Rule("import-ordering") {
 
+    private lateinit var importsLayout: String
+    private lateinit var importSorter: ImportSorter
+
+    companion object {
+        /**
+         * Alphabetical with capital letters before lower case letters (e.g. Z before a).
+         * No blank lines between major groups (android, com, junit, net, org, java, javax).
+         * Single group regardless of import type.
+         *
+         * https://developer.android.com/kotlin/style-guide#import_statements
+         */
+        private const val ASCII_PATTERN = "*"
+
+        /**
+         * Default IntelliJ IDEA style: Alphabetical with capital letters before lower case letters (e.g. Z before a),
+         * except such groups as "java", "javax" and "kotlin" that are placed in the end. Within the groups the alphabetical order is preserved.
+         * Alias imports are placed in a separate group in the end of the list with alphabetical order inside.
+         * No blank lines between groups.
+         *
+         * https://github.com/JetBrains/kotlin/blob/ffdab473e28d0d872136b910eb2e0f4beea2e19c/idea/formatter/src/org/jetbrains/kotlin/idea/core/formatter/KotlinCodeStyleSettings.java#L87-L91
+         */
+        private const val IDEA_PATTERN = "*,java.*,javax.*,kotlin.*,^*"
+
+        private const val IDEA_ERROR_MESSAGE = "Imports must be ordered in lexicographic order without any empty lines in-between " +
+            "with \"java\", \"javax\", \"kotlin\" and aliases in the end"
+        private const val ASCII_ERROR_MESSAGE = "Imports must be ordered in lexicographic order without any empty lines in-between"
+        private const val CUSTOM_ERROR_MESSAGE = "Imports must be ordered according to the pattern specified in .editorconfig"
+
+        private val errorMessages = mapOf(
+            IDEA_PATTERN to IDEA_ERROR_MESSAGE,
+            ASCII_PATTERN to ASCII_ERROR_MESSAGE
+        )
+    }
+
     override fun visit(
         node: ASTNode,
         autoCorrect: Boolean,
         emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
     ) {
+        if (node.isRoot()) {
+            val editorConfig = node.getUserData(KtLint.EDITOR_CONFIG_USER_DATA_KEY)!!
+            importsLayout = editorConfig.get("kotlin_imports_layout")
```

Comment:
I've decided to not put `ij_` in front, because it would look weird a bit if it was `ij_kotlin_imports_layout=idea`. But open for suggestions

Reply:
Hm, I think anyway it should have `ij_` prefix in the `.editorconfig` so IDEA will pickup for codestyle? Correct me if I am wrong.

If yet, I would prefer using it here as well. Could rethink this approach in the future.

Reply:
As discussed over slack, added support for both names